src/routers/DrawerRouter.js

```diff
@@ -6,6 +6,13 @@ import withDefaultValue from '../utils/withDefaultValue';
 
 import DrawerActions from './DrawerActions';
 
+const getActiveRouteKey = route => {
+  if (route.routes && route.routes[route.index]) {
+    return getActiveRouteKey(route.routes[route.index]);
+  }
+  return route.key;
```

Comment:
@ericvicenti Just a quick note here. This assumes nested navigators have globally unique route names, which may not be the case. Would it be better to use a route key that composes all the routes along the way?

For instance, my navigation stack is:
Drawer > Stack > Tab

Two different routes in the drawer can end up having routes down in the tab navigator with the same key. Tabs are just a way to filter a given page into separate areas, so they don't declare globally unique routes.

I can of course easily work around this, but thought I'd point it out in case it's causing issues for other users.

Reply:
Key is always unique for each route

Reply:
@satya164 I see. I am not using any keys anywhere, but two of my route names in nested low level navigators collide (which is not a problem for my application, cuz I never directly navigate to those routes). And this results in the following check failing: https://github.com/react-navigation/react-navigation/pull/4394/files#diff-be823f6faddbc25a62b32d94cef3e4f5R81

Either way, this is not a big deal. I can easily make those routes unique and stuff works. Feel free to ignore.

Reply:
keys are created internally